### ISSUE TITLE
ansible_galaxy corrupts filenames of git cloned repos 

### ISSUE THREAD

**Issue by vegai** [ansible/ansible#23694]
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME
ansible-galaxy

##### ANSIBLE VERSION

```
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Dec 21 2016, 07:16:46) [GCC 6.2.1 20160830]
```

##### OS / ENVIRONMENT

Arch Linux

##### SUMMARY

Filenames get corrupted  when doing ansible-galaxy install. For instance, I have a roles/filebeat/templates/platform-filebeat.service.j2 file in another repo. 

On ansible-2.2.1.0, the file is cloned correctly as platform-filebeat.service.j2.

On ansible-2.3.0.0, the filename becomes platform-.service.j2.

Looking at other files cloned like this, it strongly seems that ansible-galaxy cuts out the name of the clone ("filebeat" in the above case) from the filenames.

The YML block for the above clone looks like this (obfuscated for imagined security):

```
- src: git+ssh://gituser@gitserver/cat/ansible-role-filebeat.git
  name: filebeat
  version: master
```

**Comment by mscherer** [comment]
Related to https://github.com/ansible/ansible/pull/23633 and https://github.com/ansible/ansible/issues/22572

**Comment by hoall** [comment]
In my case a file is completly missing.

**Comment by mscherer** [comment]
yeah, I think that's the same bug. Better subscribe to https://github.com/ansible/ansible/issues/22572

**Comment by alikins** [comment]
Fix merged into devel as 79943b8
and stable-2.3 as b9ee0c5
